packages/fxa-settings/src/pages/Pair/Index/index.tsx

```diff
@@ -0,0 +1,172 @@
+  error,
+  entryPoint,
+  onSubmit,
+}: PairProps & RouteComponentProps) => {
+  usePageViewEvent(viewName, REACT_ENTRYPOINT);
+  const ftlMsgResolver = useFtlMsgResolver();
+  const localizedQRCodeLabel = ftlMsgResolver.getMsg('pair-qr-code', 'QR code');
+  const navigate = useNavigate();
+  // TODO: Recreate the QR code logic which previously existed in the content-server.
+  // Probably after that we can remove the fallback styles for the QR code div.
+
+  // TODO: check if we are either using Firefox Desktop ,or if the UA has the `supportPairing` capability
+  const isSupported = () => true;
+
+  // TODO: get the account and check if it is the default account.
+  const isDefaultAccount = () => false;
+
+  // TODO: check if the account is either not verified, or has a falsy value for session token.
+  const accountIsNotVerifiedOrHasNoSessionToken = () => false;
+
+  useEffect(() => {
+    // This will just run at page load.
+    if (!isSupported()) {
+      navigate('/pair/unsupported');
+    }
+    if (isDefaultAccount()) {
+      // we have historically needed the "forceView" option to prevent a loop of redirects.
+      navigate('/connect_another_device', { state: { forceView: true } });
+    }
+    if (accountIsNotVerifiedOrHasNoSessionToken()) {
+      navigate('/signin');
+    }
+  }, [navigate]);
+
+  const showQRCode =
+    entryPoint === ENTRYPOINTS.FIREFOX_MENU_ENTRYPOINT ||
+    entryPoint === ENTRYPOINTS.FIREFOX_PREFERENCES_ENTRYPOINT ||
+    entryPoint === ENTRYPOINTS.FIREFOX_SYNCED_TABS_ENTRYPOINT ||
+    entryPoint === ENTRYPOINTS.FIREFOX_TABS_SIDEBAR_ENTRYPOINT ||
+    entryPoint === ENTRYPOINTS.FIREFOX_FX_VIEW_ENTRYPOINT ||
+    entryPoint === ENTRYPOINTS.FIREFOX_TOOLBAR_ENTRYPOINT;
+
+  return (
+    <>
+      {showQRCode ? (
+        <CardHeader
+          headingTextFtlId="pair-sync-header"
+          headingText="Sync Firefox on your phone or tablet"
+        />
+      ) : (
+        <CardHeader
+          headingTextFtlId="pair-cad-header"
+          headingText="Connect Firefox on another device"
+        />
+      )}
+
+      <section>
+        {error && (
+          <Banner type={BannerType.error}>
+            <p>{error}</p>
+          </Banner>
+        )}
+        {showQRCode ? (
+          <>
+            <div className="mt-9 mb-5">
+              <FtlMsg id="pair-already-have-firefox">
+                <p className="text-sm">
+                  Already have Firefox on a phone or tablet?
+                </p>
+              </FtlMsg>
+            </div>
+            <div className="flex">
+              <button
+                className="cta-primary cta-xl"
+                type="button"
+                onClick={() => {
+                  onSubmit();
+                }}
+              >
+                <FtlMsg id="pair-sync-your-device-button">
+                  Sync your device
+                </FtlMsg>
+              </button>
+            </div>
+
+            <h3 className="text-lg font-bold mt-10">
```

Comment:
Looks like this skips a heading level. Should this be h2?

Reply:
Yes!! Thank you for catching this! I copied this over directly but this should be fixed